.clang-format

```diff
@@ -0,0 +1,185 @@
+# This file represents the coding style enforced by Open MPI. This file
+# is based on the long-held, but not enforced, guidelines from the
+# beginning of the project. We will be requiring that all code going
+# forward uses this style. To check your code before attempting to open
+# a PR install clang-format and run your commits through clang-format.
+#
+# To install clang-format:
+#
+# macOS:
+#   Homebrew: brew install clang-format
+#   Mac Ports: port install clang-format
+#
+# Linux:
+#   debian/ubuntu/rasbian: apt-get install clang-format
+#   redhat/fedora: yum install clang-format
+#
+# To run against your code changes:
+#
+#   unstaged changes: git clang-format --style file -f
+#   staged changes: git clang-format --style file
+#
+#   For interactive add the -p option.
+#
+# To run against all of Open MPI:
+#
+#   ./contrib/clang-format-ompi
```

Comment:
Did you forget to add this file? I don't see it anywhere on current master or in this PR.

Reply:
Still working on it. I am trying to figure out the best way to handle the excludes. Might just hard-code them into a find  command since this should just be run once.

Reply:
Added a script. Running it now. At this time it only excludes contrib and 3rd-party.